### PR TITLE
feat(autocmd): add winid in callback param table

### PR DESCRIPTION
**Pull request by glepnir**

### CODE REVIEW

**Comment by nyngwang** [comment]
@glepnir Hi, sorry for tagging you directly. Is this a try to solve #23581? I'm the author of the [comment](https://github.com/neovim/neovim/issues/23581#issuecomment-1543601599) there. I guess you know how to reproduce the bug I presented it the comment. I can help if you need someone to test. Thank you so much!

File: src/nvim/autocmd.c [hunk 1]
```diff
@@ -1972,6 +1973,7 @@ static bool call_autocmd_callback(const AutoCmd *ac, const AutoPatCmd *apc)
     PUT(data, "match", CSTR_TO_OBJ(autocmd_match));
     PUT(data, "file", CSTR_TO_OBJ(autocmd_fname));
     PUT(data, "buf", INTEGER_OBJ(autocmd_bufnr));
+    PUT(data, "winid", INTEGER_OBJ(apc->arg_winid));
```
Comment (justinmk) [inline src/nvim/autocmd.c:1976]: in sympathy with `buf` (just above this line) should this be named `win` ?
```suggestion
    PUT(data, "win", INTEGER_OBJ(apc->arg_winid));
```

**Comment by justinmk** [comment]
> when callback running the curwin may changed. so it should save the winid when autocmd triggered. 

yeah, unfortunately the semantics of "the window" [depend on the autocmd](https://github.com/neovim/neovim/issues/23581#issuecomment-1566123870). Seems like `WinClosed` would be a good test case, to start with.

**Comment by glepnir** [comment]
oh thanks I will try something with `WinClosed`.

**Pull request closed without merging**